numba/tests/test_dispatcher.py

```diff
@@ -1021,6 +1022,73 @@ def execute_with_input():
         err = execute_with_input()
         self.assertEqual(err.strip(), "cache hits = 1")
 
+    def test_user_set_cpu_name(self):
+        self.check_pycache(0)
+        mod = self.import_module()
+        mod.self_test()
+        cache_size = len(self.cache_contents())
+
+        mtimes = self.get_cache_mtimes()
+        # Change CPU name to generic
+        try:
+            os.environ['NUMBA_CPU_NAME'] = 'generic'
```

Comment:
This probably needs to have `_numba_parallel_test_ = False` as it mutates the env which is inherited by all testing threads?